drivers/triton/triton.go

```diff
@@ -0,0 +1,573 @@
+	} else {
+		// DockerApiURL looks like: 'tcp://foo.bar:2376'
+		return d.DockerApiURL, nil
+	}
+}
+
+// GetMachineName returns the name of the machine
+func (d *Driver) GetMachineName() string {
+	return d.MachineName
+}
+
+// GetSSHHostname returns hostname for use with ssh
+func (d *Driver) GetSSHHostname() (string, error) {
+	return "", fmt.Errorf("SSH is not available for the triton driver")
+}
+
+// GetSSHPort returns port for use with ssh
+func (d *Driver) GetSSHPort() (int, error) {
+	return 22, nil
+}
+
+// GetSSHUsername returns username for use with ssh
+func (d *Driver) GetSSHUsername() string {
+	return d.Account
+}
+
+// GetSSHKeyPath returns key path for use with ssh
+func (d *Driver) GetSSHKeyPath() string {
+	return path.Join(d.storePath, "id_rsa")
+}
+
+// GetURL returns a Docker compatible host URL for connecting to this host
+// e.g. tcp://1.2.3.4:2376
+func (d *Driver) GetURL() (string, error) {
+	return d.DockerApiURL, nil
+}
+
+// GetState returns the state that the host is in (running, stopped, etc)
+func (d *Driver) GetState() (state.State, error) {
+	client := &http.Client{
+		Transport: &http.Transport{
+			TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+		},
+	}
+	u := d.GetHttpsURL()
+
+	resp, err := client.Get(fmt.Sprintf("%s/_ping", u))
+	if err != nil {
+		return state.Error, nil
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+
+	if string(body[:]) == "OK" {
+		return state.Running, nil
+	}
+	return state.None, nil
+}
+
+// Kill stops a host forcefully
+func (d *Driver) Kill() error {
+	return fmt.Errorf("Kill is not available for the triton driver")
+}
+
+// PreCreateCheck allows for pre-create operations to make sure a driver is ready for creation
+func (d *Driver) PreCreateCheck() error {
+	return nil
+}
+
+// Remove a host
+func (d *Driver) Remove() error {
+	return nil
+}
+
+// Restart a host. This may just call Stop(); Start() if the provider does not
+// have any special restart behaviour.
+func (d *Driver) Restart() error {
+	return fmt.Errorf("Restart is not available for the triton driver")
+}
+
+// SetConfigFromFlags configures the driver with the object that was returned
+// by RegisterCreateFlags
+func (d *Driver) SetConfigFromFlags(flags drivers.DriverOptions) error {
+	d.CloudApiURL = flags.String("triton-url")
+	d.Account = flags.String("triton-account")
+	d.PrivateKey = flags.String("triton-key")
+
+	if d.CloudApiURL == "" {
+		return fmt.Errorf("You must specify the cloudapi url using --triton-url")
+	}
+	if d.Account == "" {
+		return fmt.Errorf("You must specify the account name using --triton-account")
+	}
+	if d.PrivateKey == "" {
+		return fmt.Errorf("You must specify the SSH key using --triton-key")
+	}
+
+	if d.CloudApiURL != "coal" && strings.IndexAny(d.CloudApiURL, ":./") == -1 {
+		// Shortend format for the cloudapi name, e.g. "us-east-3b"
+		d.CloudApiURL = fmt.Sprintf("https://%s.%s", d.CloudApiURL, TritonDefaultCloudapiDomain)
```

Comment:
Thanks for the clarification @misterbisson :)

I'm thinking maybe a generic `sdc` driver and a Joyent-specific `triton` driver would be least complicated, given all this. The `triton` driver would then take a `--triton-region` (or `--triton-datacenter`) flag, and the `sdc` driver would take a `--sdc-url` argument, and _that_ would be the place for CoaL I think.

It would be a good idea to stay somewhat in line with other CLI utilities, but it's also important for this to look and feel like the rest of the docker-machine drivers. To be clear, my main concern is that it's a little surprising for a flag named `--triton-url` to accept anything other than a real URL.